### DISCUSSION TITLE
How to remote start/stop an instance?

### DISCUSSION THREAD

**Discussion by Callahan78** [awawa-dev/HyperHDR#687] [Q&A]
Hi,
in addition to my main instance, I have two more instances for Nanoleaf panels. I would like to switch these two Nanoleaf instances on and off via ioBroker. As far as I understand, this should be possible via the JSON API. But I have no idea what I need to configure. Is this documented somewhere or are there examples?

**Comment by awawa-dev** [comment]
Hi
The "Start/Stop Instance" command in the Advanced->JSON API tab should work. You can also combine two commands (stop one instance and start the other) into one http/GET link command: the page mentioned above describes how to do this. As for ioBroker, I don't use it, so I can't help.

**Comment by Callahan78** [comment]
Meanwhile I found out how to start/stop all instances:
https://servername/json-rpc?request=%7B%22command%22:%22componentstate%22,%22componentstate%22:%7B%22component%22:%22ALL%22,%22state%22:true%7D%7D

But I didn´t find out how to start/stop a single instance. I tried to replace tge "ALL" by instance name ("Nano1") but it didn´t work.

**Comment by awawa-dev** [comment]
Thats diffent command: use the command I told you. It's in the middle of HyperHDR JSON page.

**Comment by Callahan78** [comment]
I have no "Start/Stop Instance" . . . . 
![Screenshot 2023-12-16 210209](https://github.com/awawa-dev/HyperHDR/assets/88541621/bbb63e2a-b73d-43a9-89b4-1f32522fb3db)

**Reply by awawa-dev** [reply]
you must scroll down  the page

**Comment by Callahan78** [comment]
Jesus, I was blind.
I got it and it works.
Thanks a lot!

**Comment by Stefan136** [comment]
It's somehow not working for me. I try to start the default instance 0, but it failed.

I can switch to instance 0 successfully:

{```

    "command": "instance-switchTo",
    "info": {
        "instance": 0
    },
    "success": true,
    "tan": 0
}

```
Howevever, when I try to start the instance 0, it get an error message:


```
{
    "command": "instance-startInstance",
    "error": "Can't start HyperHDR instance index 0",
    "success": false,
    "tan": 0
}
```

Stopping the instance doesn't show an error, however the instance isn't stopped 

```
{
    "command": "instance-stopInstance",
    "success": true,
    "tan": 0
}
```

Maybe I'm doing something wrong?

**Reply by awawa-dev** [reply]
it has prehistoric roots from Hyperion: you can't completely turn instance 0 off, it is always running. If you need it, move it to e.g. 1 and leave 0 unused/empty.

**Comment by Leguman31** [comment]
wrong topic sorry

**Comment by Stefan136** [comment]
I had to change the HyperHDR IP address. Since that time, the Instances List contains a link to the previous IP address. I'm wondering how to remove it? 

<img width="186" alt="Bildschirmfoto 2025-02-03 um 17 48 49" src="https://github.com/user-attachments/assets/3ee20540-2168-43e8-b090-3e2553ebf13b" />

**Reply by awawa-dev** [reply]
It is not cached by HyperHDR: it is live -discovered.

**Reply by Stefan136** [reply]
Discovered?  This IP address doesn't exist anymore. I changed the entire network from 192.168.5.x. 
Any idea where this information could come from?

**Reply by awawa-dev** [reply]
mdns

**Comment by Stefan136** [comment]
Thanks a lot for the hint. I found the mDNS advertised on the raspberry pi. Just wondering where this information is origin on the Pi?

`=    br0 IPv4 raspberrypi:8090                              _hyperhdr-http._tcp  local
   hostname = [raspberrypi.local]
   address = [192.168.5.25]
   port = [8090]
   txt = ["version=20.0.0.0"]`

The br0 interface IP as been changed - what file could still maintain the old IP? :( 

`br0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.100.25  netmask 255.255.255.0  broadcast 192.168.100.255`

**Reply by awawa-dev** [reply]
avahi and the router may cache names
